spec.html

```diff
@@ -35904,6 +35912,27 @@ <h1>Static Semantics: Early Errors</h1>
             It is a Syntax Error if the MV of the first |DecimalDigits| is strictly greater than the MV of the second |DecimalDigits|.
           </li>
         </ul>
+        <emu-grammar>Atom :: `(?` RegularExpressionModifiers `:` Disjunction `)`</emu-grammar>
+        <ul>
+          <li>
+            It is a Syntax Error if the source text matched by |RegularExpressionModifiers| contains the same code point more than once.
+          </li>
+        </ul>
+        <emu-grammar>Atom :: `(?` RegularExpressionModifiers `-` RegularExpressionModifiers `:` Disjunction `)`</emu-grammar>
+        <ul>
+          <li>
+            It is a Syntax Error if the source text matched by the first |RegularExpressionModifiers| and the source text matched by the second |RegularExpressionModifiers| are both empty.
```

Comment:
Should we just throw the early error when the second `|RegularExpressionModifiers|` is empty regardless of whether the first modifiers are empty? For example, currently the spec allows `/(?i-:foo)/`, but the implementation differs: V8 accepts this production while Babel rejects it.

Reply:
Possibly? We only error here because we decided to be more pedantic about modifiers than other languages. Other languages seem to parse `(?-:)` just fine. Any change here would need consensus.

Reply:
I remembered that I was wondering the same (why [the `start` constant is placed here](https://github.com/microsoft/TypeScript/pull/55600/files#diff-e19070e6b0ecc4de5f7d4bc6b42a9a71c8f23ce8ce155f7bd7cd48763170e506R2597)) when I was reviewing my RegExp check implementation for TypeScript.
I believe that Huáng’s expectation is actually more ergonomic, otherwise it would not be so easy for the current behavior to be misimplemented.